core/Pos/Core/Fee.hs

```diff
@@ -0,0 +1,44 @@
+module Pos.Core.Fee
+       ( Coeff(..)
+       , TxSizeLinear(..)
+       , TxFeePolicy(..)
+       ) where
+
+import           Universum
+
+import           Data.Fixed          (Fixed (..), Nano, showFixed)
+import qualified Data.Text.Buildable as Buildable
+import           Formatting          (bprint, build, shown, (%))
+
+-- | A fractional coefficient of fixed precision.
+newtype Coeff = Coeff Nano
+    deriving (Eq, Show, Generic, NFData)
+
+instance Buildable Coeff where
+    build (Coeff x) = fromString (showFixed True x)
+
+-- | A linear equation on the transaction size. Represents the @\s -> a + b*s@
+-- function where @s@ is the transaction size, @a@ and @b@ are constant
+-- coefficients.
+data TxSizeLinear = TxSizeLinear !Coeff !Coeff
```

Comment:
1. Initially the design was with a quadratic term, `a + b * txSize + c * txSize * txSize`, so I'd rather treat those coefficients numerically, without units (otherwise `c` would have to be `coin/bytes^2`.
2. Why do porpose `applyCoinPortion` if multiplication is with bytes, not coins?